### PR TITLE
WIP: add support for XGBoost inferencing

### PR DESCRIPTION
**Pull request by bfhealy**
This PR adds support for inference using trained XGB models. The code runs and is being tested on a ZTF field to ensure that DNN inference is unchanged while XGB inference runs correctly.

### CODE REVIEW

File: scope.py [hunk 1]
```diff
@@ -1010,7 +1014,7 @@ def create_inference_script(
                 algorithm = 'xgb'
                 script.write('echo "xgb inference"\n')
                 for tag in model_tags:
-                    tag_file_gen = (group_path / tag).glob('*.h5')
+                    tag_file_gen = (group_path / tag).glob('*.json')
```
Comment (mcoughlin) [inline scope.py:1017]: Should this glob both?

File: scope/xgb.py [hunk 2]
```diff
@@ -86,7 +86,7 @@ def evaluate(self, X_test, y_test, **kwargs):
         return self.model.eval(dtest, 'dtest', **kwargs)
 
     def load(self, path_model, **kwargs):
-        self.load_model(path_model, **kwargs)
+        self.model.load_model(path_model, **kwargs)
```
Comment (mcoughlin) [inline scope/xgb.py:89]: Any need to try except for both here?

**Review by mcoughlin — APPROVED** [review]
A few small questions. LGTM.

File: scope.py (hunk 1 above)
Comment (bfhealy) [reply scope.py:1017]: So far it seems XGB models are best saved as json files, and that's the only format our code saves them in. For now I think it's appropriate to only glob json extensions for the XGB algorithm.

File: scope/xgb.py (hunk 2 above)
Comment (bfhealy) [reply scope/xgb.py:89]: That's a good idea, I'll incorporate a try/except loop into this method.

**Pull request merged**